apps/site/worker.ts

```diff
@@ -1,66 +1,56 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore This file won’t exist if it hasn’t yet been built
+import * as build from "./build/server"; // eslint-disable-line import/no-unresolved
+// eslint-disable-next-line import/no-unresolved
+import __STATIC_CONTENT_MANIFEST from "__STATIC_CONTENT_MANIFEST";
 
-let remixHandler: ReturnType<typeof createRequestHandler>;
-
-const assetManifest = JSON.parse(manifestJSON);
+const MANIFEST = JSON.parse(__STATIC_CONTENT_MANIFEST);
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const handleRemixRequest = createRequestHandler(build as any as ServerBuild);
 
 export default {
-  async fetch(request: Request, env: Env, ctx: ExecutionContext) {
+  async fetch(request, env, ctx) {
+    const waitUntil = ctx.waitUntil.bind(ctx);
+    const passThroughOnException = ctx.passThroughOnException.bind(ctx);
     try {
+      const url = new URL(request.url);
+      const ttl = url.pathname.startsWith("/assets/")
+        ? 60 * 60 * 24 * 365 // 1 year
+        : 60 * 5; // 5 minutes
       return await getAssetFromKV(
-        {
-          request,
-          waitUntil(promise) {
-            return ctx.waitUntil(promise);
-          },
-        },
+        { request, waitUntil },
         {
           ASSET_NAMESPACE: env.__STATIC_CONTENT,
-          ASSET_MANIFEST: assetManifest,
+          ASSET_MANIFEST: MANIFEST,
+          cacheControl: {
+            browserTTL: ttl,
+            edgeTTL: ttl,
+          },
         }
       );
-    } catch (e) {
-      if (e instanceof NotFoundError || e instanceof MethodNotAllowedError) {
-        // fall through to the remix handler
-      } else {
-        return new Response("An unexpected error occurred", { status: 500 });
-      }
+    } catch (error) {
+      // No-op
     }
 
-    if (!remixHandler) {
-      remixHandler = createRequestHandler(build as any, process.env.NODE_ENV);
-    }
-
-    const sessionStorage = createCookieSessionStorage({
-      cookie: {
-        httpOnly: true,
-        path: "/",
-        secure: Boolean(request.url.match(/^(http|ws)s:\/\//)),
-        secrets: [env.SESSION_SECRET],
-      },
-    });
-
-    const session = await sessionStorage.getSession(
-      request.headers.get("Cookie")
-    );
-
     try {
-      return await remixHandler(request, {
-        env,
-      });
-    } catch (reason) {
-      console.error(reason);
-      return new Response("Internal Server Error", { status: 500 });
+      const loadContext = {
+        cloudflare: {
+          // This object matches the return value from Wrangler's
+          // `getPlatformProxy` used during development via Remix's
+          // `cloudflareDevProxyVitePlugin`:
+          // https://developers.cloudflare.com/workers/wrangler/api/#getplatformproxy
+          cf: request.cf,
+          ctx: { waitUntil, passThroughOnException },
+          caches,
+          env,
+        },
+      };
```

Comment:
I think we'll need to package this up somehow in an exported function to be reused inside `vite.config` for development. E.g. when running the dev script right now, it complains about lacking `auth` and `session` indicating that they're not being injected into the context properly

Reply:
this logic is actually still packaged up in the `handleFetch` util in @superflare/remix, as here: https://github.com/jplhomer/superflare/pull/66/files#diff-c48c60214196d8cfdc34b781f4a782d02027358ef29712947d121ac4b010bac9R40

but this is the worker.ts for the docs site, so it doesn’t have superflare